src/util/hass-translation.ts

```diff
@@ -64,43 +69,51 @@ export function getActiveTranslation() {
 // when DOM is created in Polymer. Even a cache lookup creates noticeable latency.
 const translations = {};
 
-export function getTranslation(fragment?, translationInput?) {
-  const translation = translationInput || getActiveTranslation();
-  const metadata = translationMetadata.translations[translation];
+export async function getTranslation(
+  fragment: string | null,
+  language: string
+) {
+  const metadata = translationMetadata.translations[language];
   if (!metadata) {
-    if (translationInput !== "en") {
+    if (language !== "en") {
       return getTranslation(fragment, "en");
     }
-    return Promise.reject(new Error("Language en not found in metadata"));
+    throw new Error("Language en is not found in metadata");
   }
   const translationFingerprint =
-    metadata.fingerprints[
-      fragment ? `${fragment}/${translation}` : translation
-    ];
+    metadata.fingerprints[fragment ? `${fragment}/${language}` : language];
 
-  // Create a promise to fetch translation from the server
+  // Fetch translation from the server
   if (!translations[translationFingerprint]) {
-    translations[translationFingerprint] = fetch(
-      `/static/translations/${translationFingerprint}`,
-      { credentials: "same-origin" }
-    )
-      .then((response) => response.json())
-      .then((data) => ({
-        language: translation,
-        data,
-      }))
-      .catch((error) => {
-        delete translations[translationFingerprint];
-        if (translationInput !== "en") {
-          // Couldn't load selected translation. Try a fall back to en before failing.
-          return getTranslation(fragment, "en");
-        }
-        return Promise.reject(error);
-      });
+    try {
+      const response = await fetch(
```

Comment:
Now this function won't finish until we are done fetching the data. It used to finish right away and it would download in parallel. 

Why would we wait till the download is done ?

Reply:
We can always extract this logic into a separate function that is async, so we can still use async/await, just not block the function further.

Reply:
It indeed has problem, even in my await version, it still has chance UI finished construction before the getTranslation promise fulfilled, there is no redraw UI or similar event fired after translation resource fetched. In that case user will get a screen missing all string resource. I need rethink this part.

Reply:
Alright, the issue is not related with my changes. In fact, current lovelace UI won't be able to switch language like the old UI does. Have to refresh (F5) the page to change the `localize` function after user selected different language.

Anyway, I changed back to then() style.

Reply:
I would consider that a bug in Lovelace, if they use the localize function, they should check if it has changed.